R/get_source_expressions.R

```diff
@@ -79,12 +79,12 @@ get_source_expressions <- function(filename, lines = NULL) {
   names(source_expression$lines) <- seq_along(source_expression$lines)
   source_expression$content <- get_content(source_expression$lines)
   parsed_content <- get_source_expression(source_expression, error = function(e) lint_parse_error(e, source_expression))
-  top_level_map <- generate_top_level_map(parsed_content)
 
-  if (inherits(e, "lint") && !nzchar(e$line)) {
+  if (inherits(e, "lint") && (is.na(e$line) || !nzchar(e$line))) {
     # Don't create expression list if it's unreliable (invalid encoding or unhandled parse error)
     expressions <- list()
   } else {
+    top_level_map <- generate_top_level_map(parsed_content)
```

Comment:
not needed in the `inherits("lint")` branch, so no need to run this code there. I _think_ it's safe to assume the `while()` loop in `generate_top_level_map()` will work as long as `e` is `NULL`, so didn't touch that. But for example, we could add a conservative test that the `while()` loop must terminate, say, at at most the # rows of the parse content.